operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -258,10 +389,31 @@ private String getLockName(String namespace, String name) {
      * @param resource The custom resource
      * @throws InvalidResourceException if the resource cannot be safely reconciled.
      */
-    protected void validate(T resource) {
+    protected List<Condition> validate(T resource) {
         if (resource != null) {
-            ResourceVisitor.visit(resource, new ValidationVisitor(resource, log));
+            Set<String> unknownFields = new HashSet<>(0);
+            Set<String> deprecatedFields = new HashSet<>(0);
+
+            ResourceVisitor.visit(resource, new ValidationVisitor(resource, log, unknownFields, deprecatedFields));
```

Comment:
Wouldn't it be easier if the `ValidationVisitor` just returned a `List<Condition>`, rather than having to map the two lists of `Strings` to `Conditions`?